frontend/helpers/schemas/allowance.ts

```diff
@@ -0,0 +1,96 @@
+import { validateAmount } from "@/utils";
+import { validatePrincipal } from "@/utils/identity";
+import * as z from "zod";
+
+export const validationMessage = {
+  spender: "Invalid principal",
+  expiration: "Invalid expiration date",
+  dateRequired: "Expiration ate is required",
+  dateNowAllowed: "Expiration Date is not allowed",
+  expiredDate: "Select a Expiration Date after the present",
+  lowBalance: "Sub account balance is not enough",
+  invalidAmount: "Amount has not a valid format",
+};
+
+export const allowanceSchema = z
```

Comment:
on allowance.ts is the same interface. why?

Reply:
Done